### PR TITLE
CircleCI -> Azure

### LINKED ISSUE #22396: CircleCI 2.0 followup
1. Set up test summary for failures
2. Document running locally with docker and `circle-ci` command

### LINKED ISSUE #22763: CI: Improve CircleCI
Not sure how much of that needs to be done upstream, but right now, CircleCI is performing very poorly. 

We're on the sixth page of jobs (so about 24h waiting time for commits coming in now), and CircleCI is stalling intermittently, e.g. right now, since about half an hour (not sure if those workers may be changing to some private pipelines that are not visible to me):

4e

Further improvements that would speed things up:
* skip jobs also if there are newer force-pushed (e.g. squashed) commits
* respect `[ci skip]` if issued
* sometimes, only 1-2 workers are running (on change-over between PRs), instead of 4
* more workers?
* etc...

### PR DESCRIPTION
**Pull request by TomAugspurger**
Closes #22396 
closes #22763 

cc @jreback @h-vetinari @jorisvandenbossche @WillAyd 

The main thing we lose is redundancy on the sql tests. Those are down to just travis.

### CODE REVIEW

File: doc/source/contributing.rst [hunk 1]
```diff
@@ -645,27 +645,18 @@ See :ref:`contributing.warnings` for more.
...
-The *pandas* test suite will run automatically on `Travis-CI <https://travis-ci.org/>`__,
-`Appveyor <https://www.appveyor.com/>`__, and `Circle CI <https://circleci.com/>`__ continuous integration
-services, once your pull request is submitted.
+The *pandas* test suite will run automatically on the `Travis-CI <https://travis-ci.org/>`__,
+and  `Azure-DevOps <https://dev.azure.com>`__ services, once your pull request is submitted.
 However, if you wish to run the test suite on a branch prior to submitting the pull request,
 then the continuous integration services need to be hooked to your GitHub repository. Instructions are here
-for `Travis-CI <http://about.travis-ci.org/docs/user/getting-started/>`__,
-`Appveyor <https://www.appveyor.com/docs/>`__ , and `CircleCI <https://circleci.com/>`__.
+for `Travis-CI <http://about.travis-ci.org/docs/user/getting-started/>`__.
```
Comment (jschendel) [inline doc/source/contributing.rst]: small nitpick: http --> https

**Comment by TomAugspurger** [comment]
To summarize the changes:

Adds 3 new azure builds

1. 2.7-compat
2. 3.6-locale
3. 3.6-local-slow

That puts us up to 6 azure builds.

The 3.5 build is dropped. I think that's fine for now.

The locale stuff isn't working yet. Still debugging.

**Comment by TomAugspurger** [comment]
OK I *think* the locale setting took effect. If you look at the summary in https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=656&view=ms.vss-test-web.test-result-details and search for a locale-related test, e.g. `test_guess_datetime_format_with_locale_specific_formats`, you'll see the skip message is `Specific locale is set zh_CN`.

I manually cancelled the travis build FYI, it didn't fail on it's own.

**Comment by jreback** [comment]
would be ok leaving the 1 circle ci test

File: ci/azure/linux.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        ci/script_multi.sh
+        echo "[Test done]"
+      displayName: 'Test'
+    - script: |
+        export PATH=$HOME/miniconda3/bin:$PATH
+        source activate pandas && pushd /tmp && python -c "import pandas; pandas.show_versions();" && popd
+    - task: PublishTestResults@2
+      inputs:
+        testResultsFiles: 'test-data-*.xml'
+        testRunTitle: 'Linux'
```
Comment (jbrockmendel) [inline ci/azure/linux.yml]: GH wants a \n here

**Comment by jbrockmendel** [comment]
Needs rebase.

If circleCI isn't completely removed, it might be nice to collect the relevant scripts in ci/circleci/

**Comment by TomAugspurger** [comment]
Restored one circle job for the SQL redundancy.

I'd recommend that if circle is running behind, we feel free to ignore it for changes that aren't likely to affect the SQL behavior.

**Comment by TomAugspurger** [comment]
This is passing.

I'm not thrilled about having to maintain the circleci scripts, but it's probably worth it for the SQL redundancy.

**Comment by jreback** [comment]
@TomAugspurger I think i merged something in contributing.rst so rebase and let's merge on green.

**Comment by TomAugspurger** [comment]
Apparently, circleCI missed the web hook on my latest one. But it did pass in https://circleci.com/gh/pandas-dev/pandas/26751?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link, which only differed by the doc fixup, so merging.

**Pull request merged**